ClientTests/Frontend/Browser/WebView/WebViewNavigationHandlerTests.swift

```diff
@@ -0,0 +1,218 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+@testable import Client
+
+import WebKit
+import XCTest
+
+class WebViewNavigationHandlerTests: XCTestCase {
+
+    // MARK: - Data scheme
+
+    func testDoesntFilterSubframes() {
+        let handler: (WKNavigationActionPolicy) -> Void = { policy in
+            XCTAssertEqual(policy, .allow, "Allows subframes")
+        }
+
+        let navigationHandler = WebViewNavigationHandlerImplementation(decisionHandler: handler)
+        let policy = WKNavigationActionMock()
+        policy.overridenTargetFrame = WKFrameInfoMock(isMainFrame: false)
+
+        navigationHandler.filterDataScheme(url: URL(string: "www.testurl.com")!,
```

Comment:
Technically non-data URLs would not be called directly with that function since we check it beforehand in the prod code. It's true my tests didn't make that implicit and wasn't ideal on that end. I made some changes to make it explicit and added a new `shouldFilterDataScheme:url` function for that.